### PR TITLE
Add Function Comment

### PR DESCRIPTION
**Pull request by makluganteng**
I added some comment into the functions in src/kakarot/account/eoa/library.cairo. 

i work on this 1 day period.

Time spent on this PR: 1 day.

## Pull request type

Added function comment into src/kakarot/account/eoa/library.cairo.

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ X] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #<Issue number> [#516]("https://github.com/sayajin-labs/kakarot/issues/516")

## What is the new behavior?

No behavior changes in this PR since its a added comments into functions.

- Added comments to the code no behaviour change.

### CODE REVIEW

File: src/kakarot/accounts/eoa/library.cairo [hunk 1]
```diff
@@ -98,6 +101,15 @@ namespace ExternallyOwnedAccount {
         );
     }
 
+    // @notice This function is to execute a function
```
Comment (ClementWalter) [inline src/kakarot/accounts/eoa/library.cairo:104]: Execute _transactions_ indeed

**Review by ClementWalter — CHANGES_REQUESTED** [review]
Thanks for your first PR!

For documentation, plz prefer descriptive syntax like
> @param a_param The X of Y

without extra verb ("takes in", etc.)

File: src/kakarot/accounts/eoa/library.cairo [hunk 2]
```diff
@@ -43,6 +43,8 @@ namespace ExternallyOwnedAccount {
     }
 
     // @notice This function is used to initialize the externally owned account.
+    // @param kakarot_address takes in a kakarot address. 
+    // @param evm_address takes in EVM address.
```
[the remark this answers is not in the document]
Comment (makluganteng) [reply src/kakarot/accounts/eoa/library.cairo:47]: is this the EVM contract address

[the remark this answers is not in the document]
Comment (makluganteng) [reply src/kakarot/accounts/eoa/library.cairo:47]: i mean the EVM address version of the kakarot address

File: src/kakarot/accounts/eoa/library.cairo [hunk 3]
```diff
@@ -43,6 +43,8 @@ namespace ExternallyOwnedAccount {
     }
 
     // @notice This function is used to initialize the externally owned account.
+    // @param kakarot_address The kakarot smart contract address. 
+    // @param evm_address The EVM address version of the kakarot address.
```
Comment (ClementWalter) [inline src/kakarot/accounts/eoa/library.cairo:47]: I think that there is something unclear to you regarding the current kakarot architecture and consequently these parameters.
In the EVM, there are two types of accounts: Contract Account (smart contract) and Externally Owned Account (addresses controlled by wallets)
In kakarot, there is a smart contract deployed for each EVM address (see `IAccount` interface`). This contract, thus a Starknet contract, has a storage var `evm_address` to store the EVM address is contract is deployed for

File: src/kakarot/accounts/eoa/library.cairo [hunk 4]
```diff
@@ -70,7 +72,7 @@ namespace ExternallyOwnedAccount {
         return (evm_address=address);
     }
 
-    // @notice Check if tx is signed and valid for each call.
+    // @notice Check if tx is signed and valid for each call and validate the transaction.
```
Comment (ClementWalter) [inline src/kakarot/accounts/eoa/library.cairo:75]: what does "validate" a tx means of not checking that the signature is valid? You could however give more details about how this is done:
- the whole EVM signed tx is given in the calldata
- it is rlp decoded, then message hash is computed and signature is checked against this hash (see `EthTransaction.validate` for more details)

**Review by ClementWalter — CHANGES_REQUESTED** [review]

**Comment by ClementWalter** [comment]
@makluganteng would be able to finish the work?

**Comment by makluganteng** [comment]
> 

yes i will, sorry iwasnt able to do it this past few weeks now ill try to finish the PR

**Review by ClementWalter — CHANGES_REQUESTED** [review]

File: src/kakarot/accounts/eoa/library.cairo [hunk 5]
```diff
@@ -43,6 +43,8 @@ namespace ExternallyOwnedAccount {
     }
 
     // @notice This function is used to initialize the externally owned account.
+    // @param kakarot_address The address of the kakarot contract
```
Comment (ClementWalter) [inline src/kakarot/accounts/eoa/library.cairo:46]: `_kakarot_address`

Comment (ClementWalter) [reply src/kakarot/accounts/eoa/library.cairo:46]: `_evm_address`

File: src/kakarot/accounts/eoa/library.cairo [hunk 6]
```diff
@@ -98,6 +100,13 @@ namespace ExternallyOwnedAccount {
         );
     }
 
+    // @notice Execute the transaction.
+    // @param call_array_len The length of the call array.
+    // @param call_array The call array.
+    // @param calldata_len The length of the calldata.
+    // @param calldata The calldata.
+    // @param response The response.
```
Comment (ClementWalter) [inline src/kakarot/accounts/eoa/library.cairo:108]: The response data array to be updated

File: src/kakarot/accounts/eoa/library.cairo [hunk 7]
```diff
@@ -98,6 +100,13 @@ namespace ExternallyOwnedAccount {
...
+    // @return response_len The length of the response.
```
Comment (ClementWalter) [inline src/kakarot/accounts/eoa/library.cairo:109]: The total length of the response data array

**Comment by ClementWalter** [comment]
@makluganteng please fir your conflicts

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**